### PR TITLE
 RMF: Add support JPEG compressed RMF datasets 

### PR DESCRIPTION
**Pull request by drons**

### CODE REVIEW

**Comment by drons** [comment]
I see some configuration changes at travis. I'm rebase my changes over master (496edf2698c929a8a45160cbdbf09a864314986d).

File: gdal/frmts/rmf/GNUmakefile [hunk 1]
```diff
@@ -1,9 +1,17 @@
 
 include ../../GDALmake.opt
 
-OBJ	=	rmfdataset.o rmflzw.o rmfdem.o
+OBJ	=	rmfdataset.o rmflzw.o rmfdem.o rmfjpeg.o
 
+JPEG_FLAGS =
```
Comment (rouault) [inline gdal/frmts/rmf/GNUmakefile]: JPEG_FLAGS useless ?

File: gdal/frmts/rmf/rmfdataset.cpp [hunk 2]
```diff
@@ -402,15 +398,27 @@ CPLErr RMFRasterBand::IReadBlock( int nBlockXOff, int nBlockYOff,
             if( nTileSize > nBlockSize )
                 nTileSize = nBlockSize;
 
-            for( GUInt32 i = 0; i < nTileSize; i++ )
+            if(poGDS->bReverseBandLayout && bCompresedTile)
```
Comment (rouault) [inline gdal/frmts/rmf/rmfdataset.cpp]: typo. should be bCompressedTile

File: gdal/frmts/rmf/rmfjpeg.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    oJpegInfo.src = &oSrc;
+    jpeg_read_header(&oJpegInfo, TRUE);
+
+    oJpegInfo.dct_method = JDCT_FLOAT;
+    oJpegInfo.out_color_space = JCS_RGB;
+
+    jpeg_start_decompress(&oJpegInfo);
+
+    JDIMENSION  nImageHeight = std::min(oJpegInfo.image_height,
```
Comment (rouault) [inline gdal/frmts/rmf/rmfjpeg.cpp]: shouldn't there be a check that the number of components is 3 ?

Comment (drons) [reply gdal/frmts/rmf/rmfjpeg.cpp]: It is not possible to create files with a number of bands other than 3.
I'm check it at this [point](https://github.com/OSGeo/gdal/pull/691/files/3a41a23f95fa3a74ce4b80baac20a230f3e52573#diff-a59133a9033503c9219f44d776451b8fR1747).
But it is better to check it here for a future.

Comment (rouault) [reply gdal/frmts/rmf/rmfjpeg.cpp]: My comment was to deal with hostile files where the JPEG codestream wouldn't match other expectations of the code

Comment (drons) [reply gdal/frmts/rmf/rmfjpeg.cpp]: Yes, it is important.

**Comment by rouault** [comment]
Looks good to me. Do you mind if I squash the commits in a single one for simpler history ?

**Pull request merged**